### PR TITLE
chore: expands menu and combobox JSX API documentation

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Adds new sections to Menu and Combobox API docs about JSX requirements to help address repeated pitfalls in component composition.

## Checklist

- ~~:ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)~~
- [ ] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- ~~:link: considered opportunities for adding cross-reference URLs (grep for keywords)~~
- ~~:wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver~~
- ~~:memo: tested in Chrome, Firefox, Safari, Edge~~

### CODE REVIEW

**Review by ze-flo — COMMENTED** [review]
Should we mention that Menu will only recognizes sub component with the same prop signature as `Item` and `ItemGroup`?

**Comment by geotrev** [comment]
> Should we mention that Menu will only recognizes sub component with the same prop signature as `Item` and `ItemGroup`?

I wondered about that too. Yeah, feels like it might be worth it since it happens so often in consumer code.

File: src/pages/components/menu.mdx [hunk 1]
```diff
@@ -207,6 +207,26 @@ const ITEMS = [
...
+
+<Menu>
+  <ItemGroupComponent legend="Flowers">
+    <ItemComponent value="Violet" />
+    <ItemComponent value="Sunflower" />
+    <ItemComponent value="Daisy" />
+  </ItemGroupComponent>
+</Menu>
+```
+
```
Comment (ze-flo) [inline src/pages/components/menu.mdx:229]: ❤️

**Review by ze-flo — APPROVED** [review]
Welcomed additions to our docs. Great job! 💯

File: src/pages/components/combobox.mdx [hunk 2]
```diff
@@ -239,11 +239,63 @@ The Combobox component follows this structure:
...
     <Option />
+    <OptGroup>
+      <Option />
+    </OptGroup>
     {/* etc. */}
   </Combobox>
 </Field>
 ```
 
+When authoring a `Combobox`, prefer either a vanilla function and/or `Array.map` to generate options:
```
Comment (jzempel) [inline src/pages/components/combobox.mdx:250]: Re-write to simplify and use [active voice](https://garden.zendesk.com/content/principles#use-the-active-voice). Something like:

```
Use `Array.map` to iterate over options.
```

Tools like https://hemingwayapp.com/ can help.

File: src/pages/components/combobox.mdx [hunk 3]
```diff
@@ -239,11 +239,63 @@ The Combobox component follows this structure:
...
+        ))}
+      </OptGroup>
+    ) : (
+      <Option key={option.value} {...option} />
+    )
+  )}
+</Combobox>
+```
+
+If wrapping subcomponents `OptGroup` or `Option`, ensure the prop signature matches the contained components:
```
Comment (jzempel) [inline src/pages/components/combobox.mdx:279]: ditto

File: src/pages/components/menu.mdx [hunk 4]
```diff
@@ -171,10 +171,62 @@ The Menu component follows this structure:
...
   <Item />
   <Item />
+  <ItemGroup>
+    <Item />
+  </ItemGroup>
   {/* etc. */}
 </Menu>
 ```
 
+When authoring a `Menu`, prefer either a vanilla function and/or `Array.map` to generate items:
```
Comment (jzempel) [inline src/pages/components/menu.mdx:181]: ditto

File: src/pages/components/menu.mdx [hunk 5]
```diff
@@ -171,10 +171,62 @@ The Menu component follows this structure:
...
+        ))}
+      </ItemGroup>
+    ) : (
+      <Item key={item.value} {...item} />
+    )
+  )}
+</Menu>
+```
+
+If wrapping subcomponents `ItemGroup` or `Item`, ensure the prop signature matches the contained components:
```
Comment (jzempel) [inline src/pages/components/menu.mdx:210]: ditto

**Review by jzempel — CHANGES_REQUESTED** [review]

**Comment by geotrev** [comment]
I realized we don't necessarily use code styling when referencing a component name. Should I remove the ticks from the new text?

Compare "Menu" in the description here:

Screenshot 2024-08-06 at 12 22 46 PM

With the new text in this PR:

Screenshot 2024-08-06 at 12 23 38 PM

File: src/pages/components/combobox.mdx [hunk 6]
```diff
@@ -239,11 +239,63 @@ The Combobox component follows this structure:
...
+Match `Option` and `OptGroup` prop signatures when using wrapper components.
+
+```jsx
+const OptionComponent = (props) => (
+  <Option {...props} />
+)
+
+const OptGroupComponent = (props) => (
+  <OptGroup {...props} />
+)
```
Comment (jzempel) [inline src/pages/components/combobox.mdx:288]: Should the canonical example be using `forwardRef`?

**Comment by jzempel** [comment]
> I realized we don't necessarily use inline code styling when referencing a component name. Should I remove the ticks from the new text?
> 
> Compare "Menu" in the description here:
> 
> Screenshot 2024-08-06 at 12 22 46 PM
> With the new text in this PR:
> 
> Screenshot 2024-08-06 at 12 23 38 PM

Within the context of the API section, the code naming and treatment makes sense.

File: src/pages/components/menu.mdx [hunk 7]
```diff
@@ -171,10 +171,62 @@ The Menu component follows this structure:
...
+Match `Item` and `ItemGroup` prop signatures when using wrapper components.
+
+```jsx
+const ItemComponent = (props) => (
+  <Item {...props} />
+)
+
+const ItemGroupComponent = (props) => (
+  <ItemGroup {...props} />
+)
```
Comment (jzempel) [inline src/pages/components/menu.mdx:219]: Should `forwardRef` be shown here, too?

**Review by jzempel — APPROVED** [review]

File: src/pages/components/menu.mdx (hunk 7 above)
Comment (geotrev) [reply src/pages/components/menu.mdx]: Thanks for catching this. :D

**Pull request merged**